FormalConjectures/ErdosProblems/1084.lean

```diff
@@ -30,24 +30,44 @@ Estimate the growth of `f_2(n)`.
 Status: open.
 -/
 
-open Finset Real
+open Finset Filter Metric Real
 open scoped EuclideanGeometry
 
 namespace Erdos1084
+variable {n : ℕ}
 
-/--
-Erdős problem 1084.
+/-- The maximal number of pairs of points which are distance 1 apart that a set of `n` 1-separated
+points in `ℝ^d` make. -/
+noncomputable def f (d n : ℕ) : ℕ :=
+  ⨆ (s : Finset (ℝ^ d)) (_ : s.card = n) (_ : IsSeparated 1 s.toSet), unitDistNum s
 
-There exists a constant `C` such that for any finite set `s` of points in `ℝ²`
-with all pairwise distances at least `1`, the number of ordered pairs of distinct
-points in `s` at distance exactly `1` is at most `C * n^{4/3}`, where `n = card s`.
--/
+/-- It is easy to check that $f_1(n) = n - 1$. -/
+@[category research solved, AMS 52]
+theorem erdos_1084_upper_d1 (n : ℕ) : f 1 n = n - 1 :=
+  sorry
+
+/-- It is easy to check that $f_2(n) < 3n$. -/
+@[category research solved, AMS 52]
+theorem erdos_1084_easy_upper_d2 (hn : n ≠ 0) : f 2 n < 3 * n :=
+  sorry
+
+/-- Erdős showed that there is some constant $c > 0$ such that $f_2(n) < 3n - c n^{1/2}$. -/
+@[category research solved, AMS 52]
+theorem erdos_1084_upper_d2 : ∃ c > (0 : ℝ), f 2 n < 3 * n - c * sqrt n :=
```

Comment:
Can this `c` now depend on `n` the way it is written? If yes we should switch order. Perhaps it is best to inline the `variable {n  : ℕ}` from above in all places for simplicity/clarity?